src/python/pants/core/goals/lint.py

```diff
@@ -182,7 +212,7 @@ async def lint(
     union_membership: UnionMembership,
     dist_dir: DistDir,
 ) -> Lint:
-    request_types = cast("Iterable[type[StyleRequest]]", union_membership[LintRequest])
+    request_types = cast("Iterable[type[LintRequest]]", union_membership[LintRequest])
```

Comment:
You should be able to get rid of this `cast`. `union_membership[LintRequest]` already gives you back this type. 

(I thought it was intentional why we needed to use `StyleRequest`, but can't remember why that was.)

Reply:
Tried it, but unfortunately mypy isn't pleased.